### ISSUE TITLE
-Wenum-conversion in drivers/net/ethernet/qlogic/qede/qede_main.c

### ISSUE THREAD

**Issue by nathanchance** [ClangBuiltLinux/linux#326]
```
drivers/net/ethernet/qlogic/qede/qede_main.c:1183:29: warning: implicit conversion from enumeration type 'enum qede_rdma_probe_mode' to different enumeration type 'enum qede_rdma_remove_mode' [-Wenum-conversion]
        qede_rdma_dev_remove(edev, QEDE_RDMA_PROBE_MODE(mode));
        ~~~~~~~~~~~~~~~~~~~~       ^~~~~~~~~~~~~~~~~~~~~~~~~~
drivers/net/ethernet/qlogic/qede/qede_main.c:1074:33: note: expanded from macro 'QEDE_RDMA_PROBE_MODE'
        ((mode) == QEDE_PROBE_NORMAL ? QEDE_RDMA_PROBE_NORMAL \
                                       ^~~~~~~~~~~~~~~~~~~~~~
drivers/net/ethernet/qlogic/qede/qede_main.c:1183:29: warning: implicit conversion from enumeration type 'enum qede_rdma_probe_mode' to different enumeration type 'enum qede_rdma_remove_mode' [-Wenum-conversion]
        qede_rdma_dev_remove(edev, QEDE_RDMA_PROBE_MODE(mode));
        ~~~~~~~~~~~~~~~~~~~~       ^~~~~~~~~~~~~~~~~~~~~~~~~~
drivers/net/ethernet/qlogic/qede/qede_main.c:1075:12: note: expanded from macro 'QEDE_RDMA_PROBE_MODE'
                                     : QEDE_RDMA_PROBE_RECOVERY)
                                       ^~~~~~~~~~~~~~~~~~~~~~~~
2 warnings generated.
```

Patch sent: https://lore.kernel.org/lkml/20190124162700.19989-1-natechancellor@gmail.com/

**Comment by nickdesaulniers** [comment]
thanks @nathanchance ! KernelCI has noticed this, too.

**Comment by nathanchance** [comment]
@nickdesaulniers is there a place I can see the KernelCI Clang results? I don't see it on kernelci.org (at least not clearly).

**Comment by vo4** [comment]
Thanks @nathanchance for the patch. 
Nick is referring to linaro's CI https://ci.linaro.org/view/tcwg_kernel-llvm/ . We get email digests of the runs. One of them picked up the failure.

**Comment by nathanchance** [comment]
@vo4 thanks for the link, I'll check it out!

**Comment by nathanchance** [comment]
This has been fixed as of `next-20190129`. David Miller applied the series that introduced this error to `net` instead of `net-next` so he reverted it and told them to resend it and address this warning when doing so.

https://lore.kernel.org/netdev/20190127.101729.1774234996114932306.davem@davemloft.net/

https://lore.kernel.org/netdev/20190128172756.27361-1-michal.kalderon@cavium.com/
